### ISSUE TITLE
Finetuning details

### ISSUE THREAD

**Issue by asahinyener** [facebookresearch/xcit#9]
Hello; this is a great work. I would like to take advantage of the models lower vram requirement to deploy these models on edge.

However i would like to ask for resources on how to finetune the models with our data. Does finetuning follow the standart model of replacing the classification head (the final connected layer maybe?); and then applying training with a lower learning rate (what would you advise as a general baseline?).

Thanks in advance for any pointers, again great work!

**Comment by asahinyener** [comment]
I have with further examination have deduced a couple of required  changes; which is based on the mmdetection library tutorials.

Base tutorial can be found at : https://mmdetection.readthedocs.io/en/latest/tutorials/customize_dataset.html?highlight=classes

Changes to model: https://github.com/facebookresearch/xcit/blob/0e6589eef0ae3f325be53228d119fe97f7a92882/detection/configs/_base_/models/mask_rcnn_xcit_p16.py#L49
https://github.com/facebookresearch/xcit/blob/0e6589eef0ae3f325be53228d119fe97f7a92882/detection/configs/_base_/models/mask_rcnn_xcit_p16.py#L68

The number of classes are mentioned here; and should be changed according to the current training job. 

Changes to the base data type:

First we should define the new names of the classes and their length and create a tuple variable that holds these.


Starting at this line there is train,test and val definitions 

https://github.com/facebookresearch/xcit/blob/0e6589eef0ae3f325be53228d119fe97f7a92882/detection/configs/_base_/datasets/coco_detection.py#L32

They should include 

```
classes=custom_classes # previously created tuple
```



These should provide the required architectural changes. However after this; the required parameters and what they depend on i leave to the authors of this work

**Comment by aelnouby** [comment]
Thanks for your question. For image classification, finetuning requires replacing the linear classification head (https://github.com/facebookresearch/xcit/blob/0e6589eef0ae3f325be53228d119fe97f7a92882/main.py#L274). We typically use a smaller learning rate LR=5e-5.

For object detection as you have detailed, you can typically follow the mmdet guidlines for finetuning. We have not attempted any finetuning for object detection or semantic segmentation, so I don't have a clear recommendation for learning rate, but typically it should be reduced as well.
